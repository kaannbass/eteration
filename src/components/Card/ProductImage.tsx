import React from 'react';
import { View, StyleSheet } from 'react-native';
import { IconButton } from 'react-native-paper';
import FastImage from 'react-native-fast-image';

interface ProductImageProps {
  imageUri: string;
  isFavorite: boolean;
  onToggleFavorite: () => void;
  iconVisible: boolean;
}

const ProductImage: React.FC<ProductImageProps> = ({
  imageUri,
  isFavorite,
  onToggleFavorite,
  iconVisible,
}) => (
  <View style={styles.imageContainer}>
    <FastImage
      style={styles.image}
      source={{
        uri: imageUri,
        priority: FastImage.priority.high,
      }}
      resizeMode={FastImage.resizeMode.cover}
      testID="product-image"
    />
    {iconVisible && (
      <IconButton
        icon={isFavorite ? 'heart' : 'heart-outline'}
        iconColor={isFavorite ? '#e91e63' : '#ffffff'}
        size={30}
        onPress={onToggleFavorite}
        style={styles.favoriteIcon}
        testID="favorite-icon"
      />
    )}
  </View>
);

export default ProductImage;

const styles = StyleSheet.create({
  imageContainer: {
    position: 'relative',
    height: 200,
    marginBottom: 10,
  },
  image: {
    width: '100%',
    height: '100%',
    resizeMode: 'cover',
    borderRadius: 5,
  },
  favoriteIcon: {
    position: 'absolute',
    top: 0,
    right: 0,
  },
});
