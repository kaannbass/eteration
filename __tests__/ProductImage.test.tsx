import React from 'react';
import { render } from '@testing-library/react-native';
import ProductImage from '../src/components/Card/ProductImage';

describe('ProductImage Component', () => {
  const mockOnToggleFavorite = jest.fn();

  it('should match snapshot when isFavorite is true', () => {
    const { toJSON } = render(
      <ProductImage 
        imageUri="http://example.com/image.jpg"
        isFavorite={true}
        onToggleFavorite={mockOnToggleFavorite}
        iconVisible={true}
      />
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('should match snapshot when isFavorite is false', () => {
    const { toJSON } = render(
      <ProductImage 
        imageUri="http://example.com/image.jpg"
        isFavorite={false}
        onToggleFavorite={mockOnToggleFavorite}
        iconVisible={true}
      />
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
