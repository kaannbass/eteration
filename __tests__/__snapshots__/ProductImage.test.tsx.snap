// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage Component should match snapshot when isFavorite is false 1`] = `
<View
  style={
    {
      "height": 200,
      "marginBottom": 10,
      "position": "relative",
    }
  }
>
  <Image
    resizeMode="cover"
    source={
      {
        "priority": "high",
        "uri": "http://example.com/image.jpg",
      }
    }
    style={
      {
        "borderRadius": 5,
        "height": "100%",
        "resizeMode": "cover",
        "width": "100%",
      }
    }
    testID="product-image"
  />
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "transparent",
        "borderRadius": 23,
        "height": 46,
        "margin": 6,
        "position": "absolute",
        "right": 0,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
        "top": 0,
        "width": 46,
      }
    }
    testID="favorite-icon-container-outer-layer"
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "transparent",
          "borderColor": "rgba(121, 116, 126, 1)",
          "borderRadius": 23,
          "borderWidth": 0,
          "elevation": 0,
          "flex": 1,
          "overflow": "hidden",
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
        }
      }
      testID="favorite-icon-container"
    >
      <View
        accessibilityComponentType="button"
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityTraits="button"
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        centered={true}
        collapsable={false}
        focusable={true}
        hitSlop={
          {
            "bottom": 6,
            "left": 6,
            "right": 6,
            "top": 6,
          }
        }
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "overflow": "hidden",
            },
            [
              {
                "alignItems": "center",
                "flexGrow": 1,
                "justifyContent": "center",
              },
              {
                "borderRadius": 23,
              },
            ],
          ]
        }
        testID="favorite-icon"
      >
        <Text
          accessibilityElementsHidden={true}
          allowFontScaling={false}
          importantForAccessibility="no-hide-descendants"
          pointerEvents="none"
          selectable={false}
          style={
            [
              {
                "color": "#ffffff",
                "fontSize": 30,
              },
              [
                {
                  "lineHeight": 30,
                  "transform": [
                    {
                      "scaleX": 1,
                    },
                  ],
                },
                {
                  "backgroundColor": "transparent",
                },
              ],
              {
                "fontFamily": "Material Design Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          󰋕
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`ProductImage Component should match snapshot when isFavorite is true 1`] = `
<View
  style={
    {
      "height": 200,
      "marginBottom": 10,
      "position": "relative",
    }
  }
>
  <Image
    resizeMode="cover"
    source={
      {
        "priority": "high",
        "uri": "http://example.com/image.jpg",
      }
    }
    style={
      {
        "borderRadius": 5,
        "height": "100%",
        "resizeMode": "cover",
        "width": "100%",
      }
    }
    testID="product-image"
  />
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "transparent",
        "borderRadius": 23,
        "height": 46,
        "margin": 6,
        "position": "absolute",
        "right": 0,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
        "top": 0,
        "width": 46,
      }
    }
    testID="favorite-icon-container-outer-layer"
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "transparent",
          "borderColor": "rgba(121, 116, 126, 1)",
          "borderRadius": 23,
          "borderWidth": 0,
          "elevation": 0,
          "flex": 1,
          "overflow": "hidden",
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
        }
      }
      testID="favorite-icon-container"
    >
      <View
        accessibilityComponentType="button"
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityTraits="button"
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        centered={true}
        collapsable={false}
        focusable={true}
        hitSlop={
          {
            "bottom": 6,
            "left": 6,
            "right": 6,
            "top": 6,
          }
        }
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "overflow": "hidden",
            },
            [
              {
                "alignItems": "center",
                "flexGrow": 1,
                "justifyContent": "center",
              },
              {
                "borderRadius": 23,
              },
            ],
          ]
        }
        testID="favorite-icon"
      >
        <Text
          accessibilityElementsHidden={true}
          allowFontScaling={false}
          importantForAccessibility="no-hide-descendants"
          pointerEvents="none"
          selectable={false}
          style={
            [
              {
                "color": "#e91e63",
                "fontSize": 30,
              },
              [
                {
                  "lineHeight": 30,
                  "transform": [
                    {
                      "scaleX": 1,
                    },
                  ],
                },
                {
                  "backgroundColor": "transparent",
                },
              ],
              {
                "fontFamily": "Material Design Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          󰋑
        </Text>
      </View>
    </View>
  </View>
</View>
`;
